src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1700,7 +1700,7 @@ private function registerCacheConfiguration(array $config, ContainerBuilder $con
 
             if (!$container->getParameter('kernel.debug')) {
                 $propertyAccessDefinition->setFactory([PropertyAccessor::class, 'createCache']);
-                $propertyAccessDefinition->setArguments([null, null, $version, new Reference('logger', ContainerInterface::IGNORE_ON_INVALID_REFERENCE)]);
+                $propertyAccessDefinition->setArguments([null, 0, $version, new Reference('logger', ContainerInterface::IGNORE_ON_INVALID_REFERENCE)]);
```

Comment:
If that change is necessary, your `int` type-hint was a small breaking change. We could make the type-hint nullable, but I also prefer `int` over `?int` in that case.

I'd suggest that we backport this change to FrameworkBundle 4.4 along with a deprecation in `PropertyAccessor::createCache()` that is triggered if `null` is passed as `$defaultLifetime`.